library/Fission/User/CID/Mutation.hs

```diff
@@ -4,63 +4,64 @@ module Fission.User.CID.Mutation
   , createX
   ) where
 
-import Database.Selda
+import Database.Esqueleto
 import RIO.List ((\\))
-import RIO.Orphans ()
-
-import Fission.Prelude
-import Fission.Timestamp      as Timestamp
 import Network.IPFS.CID.Types as IPFS.CID
 
-import           Fission.User           (User)
-import           Fission.User.CID.Query
-import qualified Fission.User.CID.Table as Table
-import           Fission.User.CID.Types
+import Fission.Prelude
+import Fission.Models
 
 -- | Create a new, timestamped entry
-create
-  :: ( MonadRIO   cfg m
-     , HasLogFunc cfg
-     , MonadSelda     m
-     , MonadMask      m
-     )
-  => ID User
+create ::
+  ( MonadLogger m
+  , MonadTime   m
+  , MonadDB     m
+  )
+  => UserId
   -> CID
-  -> m (Maybe (ID UserCID))
-create userID (CID hash) = do
-  now   <- liftIO getCurrentTime
-  mayID <- insertUnless Table.userCIDs (eqUserCID userID hash)
-            [UserCID def userID hash <@ now]
+  -> m (Maybe UserCIDId)
+create userId cid = runDBNow \now -> do
+  mayID <- insertUnique UserCID
+    { userCIDUserFK     = userId
+    , userCIDCid        = cid
+    , userCIDInsertedAt = now
+    , userCIDModifiedAt = now
+    }
```

Comment:
Figured that this would read better

Reply:
Also note that the fields are all now prefixed by the table's name. That's generated by Persistent.

Reply:
FYI, that was one of the reasons I did the separate models, so we didn't need to have the prefixed record fields (ie. you can disable the prefixes).

Reply:
reads so nice w/ the record syntax & `runDBNow`